source/core/TypeAndDataManager.cpp

```diff
@@ -1676,20 +1676,36 @@ TypeRef ArrayType::GetSubElementAddressFromPath(SubString* path, void *start, vo
         SubString pathHead;
         SubString pathTail;
         path->SplitString(&pathHead, &pathTail, '.');
+        *end = null;
 
-        gPlatform.IO.Print(" Using array indexes in paths\n");
-
-        // If the path has a tail it needs to index the array.
-        // There may be more than one way to do so raw1d indexes, or multidim
         // may allow end point relative as well ???
-        if (pathTail.Length()) {
-            subType = GetSubElementAddressFromPath(path, start, end, allowDynamic);
-        } else {
-            // TODO parse indexes.
-            // Variable sized arrays can only be indexed if allowDynamic is true.
-            subType = null;
+        if (allowDynamic) { // Variable sized arrays can only be indexed if allowDynamic is true.
+            IntMax index = 0;
+            if (pathHead.ReadInt(&index) && index >= 0 && index < array->Length()) {
+                ArrayDimensionVector dimIndex;
+                IntIndex rank = array->Rank(), dim = 0;
+                if (pathHead.SplitString(NULL, &pathHead, ',')) {
+                    dimIndex[rank-1 - dim++] = IntIndex(index);
+                    while (dim < rank && pathHead.ReadInt(&index)) {
+                        dimIndex[rank-1 - dim++] = IntIndex(index);
+                        if (!pathHead.SplitString(NULL, &pathHead, ','))
+                            break;
+                    }
+                    while (dim < rank)
+                        dimIndex[rank-1 - dim++] = 0;
+                    if (pathHead.Length() == 0) {
+                        subType = array->ElementType();
+                        *end = (AQBlock1*)array->BeginAtND(rank, dimIndex);
+                    }
+                } else {
+                    subType = array->ElementType();
+                    *end = (AQBlock1*)array->BeginAt(IntIndex(index));
+                }
+            }
+        }
+        if (*end && pathTail.Length()) {
```

Comment:
Actually, the > 0 in the other case was because of a previous edit where it was temporarily > 1 in order to help debug an issue.  When I changed it back by hand I changed it to > 0 instead of removing the inequality.  I have reverted that edit in the other place now.